llama-index-core/llama_index/core/workflow/events.py

```diff
@@ -68,7 +74,8 @@ def __init__(self, **params: Any):
         super().__init__(**fields)
         for private_attr, value in private_attrs.items():
             super().__setattr__(private_attr, value)
-        self._data = data
+        if data:
```

Comment:
Note: the `super().__init__(...)` call sets the PrivateAttr `_data` since we defined it's `default_factory`.